File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/entity/SaleDocumentEntity.ts
import { AuditTableEntity } from "src/app/enterprise/shared/model/entity/AuditTableEntity";
import { ClientEntity } from "src/app/enterprise/client/model/entity/ClientEntity";

export class SaleDocumentEntity extends AuditTableEntity{

    public DocumentCod : string;
    public CounterfoilCod : string;
    public SaleCod : string;
    public DocumentType : string;
    public DocumentRole : string;
    public ClientCod : string;
    public IssueDate : Date | null;
    public Client : ClientEntity;

    constructor(){
        super();
        this.DocumentCod = "";
        this.CounterfoilCod = "";
        this.SaleCod = "";
        this.DocumentType = "";
        this.DocumentRole = "";
        this.ClientCod = "";
        this.IssueDate = null;
        this.Client = new ClientEntity();
    }
}

File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/entity/SaleDocumentEntity.java
package com.ccadmin.app.sale.model.entity;

import com.ccadmin.app.client.model.entity.ClientEntity;
import com.ccadmin.app.sale.model.entity.id.SaleDocumentID;
import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;

import java.io.Serializable;
import java.util.Date;

@Entity
@Table( name = "sale_document" )
@IdClass(SaleDocumentID.class)
public class SaleDocumentEntity extends AuditTableEntity implements Serializable {

    @Id
    public String DocumentCod;
    @Id
    public String SaleCod;
    public String CounterfoilCod;
    public String DocumentType;
    public String DocumentRole;
    public String ClientCod;
    public Date IssueDate;

    @Transient
    public ClientEntity Client;

}
